### PR TITLE
Fixing assembly and file versions for net core projects

### PR DESCRIPTION
**Pull request by mishra14**
Currently net core projects are using auto generated build numbers. this PR fixes that.

### CODE REVIEW

File: build/common.props [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <Project ToolsVersion="15.0">
   <PropertyGroup>
     <IsNetCoreProject>true</IsNetCoreProject>
+    <GenerateAssemblyInfo>true</GenerateAssemblyInfo>
```
Comment (emgarten) [inline build/common.props]: Move this to common.project.props with a condition.

**Review by emgarten — CHANGES_REQUESTED** [review]
Move GenerateAssemblyInfo

**Review by emgarten — APPROVED** [review]
Good fix 🚀

File: build/common.project.props [hunk 2]
```diff
@@ -169,6 +168,11 @@
     </AssemblyAttributes>
   </ItemGroup>
 
+  <!-- Assembly attributes for net core projects -->
+  <PropertyGroup Condition=" '$(IsNetCoreProject)' == 'true' ">
+    <AssemblyVersion>$(SemanticVersion).$(PreReleaseVersion)</AssemblyVersion>
```
Comment (alpaix) [inline build/common.project.props]: Why do you need to set this explicitly? Having `$(Version)` should be enough to make msbuild set correct assembly version.

**Review by alpaix — CHANGES_REQUESTED** [review]
Address the feedback.

File: build/common.project.props [hunk 3]
```diff
@@ -169,6 +168,11 @@
     </AssemblyAttributes>
   </ItemGroup>
 
+  <!-- Assembly attributes for net core projects -->
+  <PropertyGroup Condition=" '$(IsNetCoreProject)' == 'true' ">
```
Comment (nkolev92) [inline build/common.project.props]: Merge this propertygroup with the one below it?

File: build/common.project.props (hunk 2 above)
Comment (mishra14) [reply build/common.project.props]: `$(version)` alone does not work. We already had that.

**Comment by mishra14** [comment]
🔔

File: build/common.project.props [hunk 4]
```diff
@@ -170,6 +169,9 @@
   </ItemGroup>
 
   <PropertyGroup Condition=" '$(IsNetCoreProject)' == 'true' ">
+    <!-- Assembly attributes for net core projects -->
+    <AssemblyVersion>$(SemanticVersion).$(PreReleaseVersion)</AssemblyVersion>
+
```
Comment (nkolev92) [inline build/common.project.props:174]: nit: no need for the empty line.

**Review by nkolev92 — APPROVED** [review]

File: build/common.project.props (hunk 4 above)
Comment (mishra14) [reply build/common.project.props:174]: That keeps the separate logical properties separate. I think this is fine.

File: build/common.project.props (hunk 2 above)
Comment (alpaix) [reply build/common.project.props]: I created a simple netstandard library
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
    <Version>2.3.4-beta1-1256</Version>
    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
  </PropertyGroup>

</Project>
```
And it did set version correctly. What's wrong with our projects?

**Review by alpaix — APPROVED** [review]

**Pull request merged**